lib/tasks/status.rake

```diff
@@ -26,3 +26,14 @@ namespace :redis do
     end
   end
 end
+
+namespace :kafka do
+  desc 'Check for available kafka connection'
+  task status: [:environment] do
+    begin
+      TestProducer.deliver
```

Comment:
Observation: On first use this errors out.

Log:
```
[2021-06-15 07:28:26,362] INFO Creating topic test with configuration {} and initial partition assignment HashMap(0 -> ArrayBuffer(1)) (kafka.zk.AdminZkClient)
[2021-06-15 07:28:26,371] INFO [KafkaApi-1] Auto creation of topic test with 1 partitions and replication factor 1 is successful (kafka.server.KafkaApis)
[2021-06-15 07:28:26,374] INFO [Controller id=1] New topics: [HashSet(test)], deleted topics: [HashSet()], new partition replica assignment [Map(test-0 -> ReplicaAssignment(replicas=1, addingReplicas=, removingReplicas=))] (kafka.controller.KafkaController)
[2021-06-15 07:28:26,374] INFO [Controller id=1] New partition creation callback for test-0 (kafka.controller.KafkaController)
[2021-06-15 07:28:26,374] INFO [Controller id=1 epoch=1] Changed partition test-0 state from NonExistentPartition to NewPartition with assigned replicas 1 (state.change.logger)
[2021-06-15 07:28:26,374] INFO [Controller id=1 epoch=1] Sending UpdateMetadata request to brokers HashSet() for 0 partitions (state.change.logger)
[2021-06-15 07:28:26,374] TRACE [Controller id=1 epoch=1] Changed state of replica 1 for partition test-0 from NonExistentReplica to NewReplica (state.change.logger)
[2021-06-15 07:28:26,374] INFO [Controller id=1 epoch=1] Sending UpdateMetadata request to brokers HashSet() for 0 partitions (state.change.logger)
[2021-06-15 07:28:26,383] INFO [Controller id=1 epoch=1] Changed partition test-0 from NewPartition to OnlinePartition with state LeaderAndIsr(leader=1, leaderEpoch=0, isr=List(1), zkVersion=0) (state.change.logger)
[2021-06-15 07:28:26,383] TRACE [Controller id=1 epoch=1] Sending become-leader LeaderAndIsr request LeaderAndIsrPartitionState(topicName='test', partitionIndex=0, controllerEpoch=1, leader=1, leaderEpoch=0, isr=[1], zkVersion=0, replicas=[1], addingReplicas=[], removingReplicas=[], isNew=true) to broker 1 for partition test-0 (state.change.logger)
[2021-06-15 07:28:26,383] INFO [Controller id=1 epoch=1] Sending LeaderAndIsr request to broker 1 with 1 become-leader and 0 become-follower partitions (state.change.logger)
[2021-06-15 07:28:26,383] INFO [Controller id=1 epoch=1] Sending UpdateMetadata request to brokers HashSet(1) for 1 partitions (state.change.logger)
[2021-06-15 07:28:26,384] TRACE [Controller id=1 epoch=1] Changed state of replica 1 for partition test-0 from NewReplica to OnlineReplica (state.change.logger)
[2021-06-15 07:28:26,384] INFO [Controller id=1 epoch=1] Sending UpdateMetadata request to brokers HashSet() for 0 partitions (state.change.logger)
[2021-06-15 07:28:26,386] INFO [RequestSendThread controllerId=1] Controller 1 connected to localhost:29092 (id: 1 rack: null) for sending state change requests (kafka.controller.RequestSendThread)
[2021-06-15 07:28:26,386] INFO [Broker id=1] Handling LeaderAndIsr request correlationId 13 from controller 1 for 1 partitions (state.change.logger)
[2021-06-15 07:28:26,386] TRACE [Broker id=1] Received LeaderAndIsr request LeaderAndIsrPartitionState(topicName='test', partitionIndex=0, controllerEpoch=1, leader=1, leaderEpoch=0, isr=[1], zkVersion=0, replicas=[1], addingReplicas=[], removingReplicas=[], isNew=true) correlation id 13 from controller 1 epoch 1 (state.change.logger)
[2021-06-15 07:28:26,387] TRACE [Broker id=1] Handling LeaderAndIsr request correlationId 13 from controller 1 epoch 1 starting the become-leader transition for partition test-0 (state.change.logger)
[2021-06-15 07:28:26,387] INFO [ReplicaFetcherManager on broker 1] Removed fetcher for partitions Set(test-0) (kafka.server.ReplicaFetcherManager)
[2021-06-15 07:28:26,387] INFO [Broker id=1] Stopped fetchers as part of LeaderAndIsr request correlationId 13 from controller 1 epoch 1 as part of the become-leader transition for 1 partitions (state.change.logger)
[2021-06-15 07:28:26,390] INFO [Log partition=test-0, dir=/var/lib/kafka/data] Loading producer state till offset 0 with message format version 2 (kafka.log.Log)
[2021-06-15 07:28:26,390] INFO Created log for partition test-0 in /var/lib/kafka/data/test-0 with properties {compression.type -> producer, min.insync.replicas -> 1, message.downconversion.enable -> true, segment.jitter.ms -> 0, cleanup.policy -> [delete], flush.ms -> 9223372036854775807, retention.ms -> 604800000, segment.bytes -> 1073741824, flush.messages -> 9223372036854775807, message.format.version -> 2.7-IV2, max.compaction.lag.ms -> 9223372036854775807, file.delete.delay.ms -> 60000, max.message.bytes -> 1048588, min.compaction.lag.ms -> 0, message.timestamp.type -> CreateTime, preallocate -> false, index.interval.bytes -> 4096, min.cleanable.dirty.ratio -> 0.5, unclean.leader.election.enable -> false, retention.bytes -> -1, delete.retention.ms -> 86400000, segment.ms -> 604800000, message.timestamp.difference.max.ms -> 9223372036854775807, segment.index.bytes -> 10485760}. (kafka.log.LogManager)
[2021-06-15 07:28:26,391] INFO [Partition test-0 broker=1] No checkpointed highwatermark is found for partition test-0 (kafka.cluster.Partition)
[2021-06-15 07:28:26,391] INFO [Partition test-0 broker=1] Log loaded for partition test-0 with initial high watermark 0 (kafka.cluster.Partition)
[2021-06-15 07:28:26,391] INFO [Broker id=1] Leader test-0 starts at leader epoch 0 from offset 0 with high watermark 0 ISR [1] addingReplicas [] removingReplicas []. Previous leader epoch was -1. (state.change.logger)
[2021-06-15 07:28:26,394] TRACE [Broker id=1] Completed LeaderAndIsr request correlationId 13 from controller 1 epoch 1 for the become-leader transition for partition test-0 (state.change.logger)
[2021-06-15 07:28:26,395] INFO [Broker id=1] Finished LeaderAndIsr request in 9ms correlationId 13 from controller 1 for 1 partitions (state.change.logger)
[2021-06-15 07:28:26,395] TRACE [Controller id=1 epoch=1] Received response {error_code=0,partition_errors=[{topic_name=test,partition_index=0,error_code=0,_tagged_fields={}}],_tagged_fields={}} for request LEADER_AND_ISR with correlation id 13 sent to broker localhost:29092 (id: 1 rack: null) (state.change.logger)
[2021-06-15 07:28:26,396] TRACE [Broker id=1] Cached leader info UpdateMetadataPartitionState(topicName='test', partitionIndex=0, controllerEpoch=1, leader=1, leaderEpoch=0, isr=[1], zkVersion=0, replicas=[1], offlineReplicas=[]) for partition test-0 in response to UpdateMetadata request sent by controller 1 epoch 1 with correlation id 14 (state.change.logger)
[2021-06-15 07:28:26,396] INFO [Broker id=1] Add 1 partitions and deleted 0 partitions from metadata cache in response to UpdateMetadata request sent by controller 1 epoch 1 with correlation id 14 (state.change.logger)
[2021-06-15 07:28:26,396] TRACE [Controller id=1 epoch=1] Received response {error_code=0,_tagged_fields={}} for request UPDATE_METADATA with correlation id 14 sent to broker localhost:29092 (id: 1 rack: null) (state.change.logger)
```

Reply:
Yes, I believe the rake task will show 'LeaderNotAvailable' the first time. I'm not really sure how to work around it other than re-trying and being successful the second time. I don't think that should cause any issues, though, since the probe would be set to run every ~5 seconds or so.